services/ui-src/src/components/reports/ModalDrawerReportPage.tsx

```diff
@@ -37,53 +76,138 @@ export const ModalDrawerReportPage = ({ route }: Props) => {
     deleteEntityModalOnOpenHandler();
   };
 
+  const closeDeleteEntityModal = () => {
+    setSelectedEntity(undefined);
+    deleteEntityModalOnCloseHandler();
+  };
+
+  const onSubmit = async (formData: AnyObject) => {
+    if (userIsStateUser || userIsStateRep) {
+      setSubmitting(true);
+      const reportKeys = {
+        state: state,
+        id: report?.id,
+      };
+      const currentEntities = [...(report?.fieldData[entityType] || {})];
+      const selectedEntityIndex = report?.fieldData[entityType].findIndex(
+        (entity: EntityShape) => entity.id === selectedEntity?.id
+      );
+      const newEntity = {
+        ...selectedEntity,
+        ...formData,
+      };
+      let newEntities = currentEntities;
+      newEntities[selectedEntityIndex] = newEntity;
+      const dataToWrite = {
+        status: ReportStatus.IN_PROGRESS,
+        lastAlteredBy: full_name,
+        fieldData: {
+          [entityType]: newEntities,
+        },
+      };
+      await updateReport(reportKeys, dataToWrite);
+      setSubmitting(false);
+    }
+    drawerOnCloseHandler();
+  };
+
+  const getFormattedEntityData = (entity?: EntityShape) => ({
+    category: entity?.accessMeasure_generalCategory[0].value,
+    standardDescription: entity?.accessMeasure_standardDescription,
+    standardType:
+      entity?.accessMeasure_standardType[0].value !== "Other, specify"
+        ? entity?.accessMeasure_standardType[0].value
+        : entity?.["accessMeasure_standardType-otherText"],
+    provider:
+      entity?.accessMeasure_providerType?.[0].value !== "Other, specify"
+        ? entity?.accessMeasure_providerType?.[0].value
+        : entity?.["accessMeasure_providerType-otherText"],
+    region:
+      entity?.accessMeasure_applicableRegion?.[0].value !== "Other, specify"
+        ? entity?.accessMeasure_applicableRegion?.[0].value
+        : entity?.["accessMeasure_applicableRegion-otherText"],
+    population:
+      entity?.accessMeasure_population?.[0].value !== "Other, specify"
+        ? entity?.accessMeasure_population?.[0].value
+        : entity?.["accessMeasure_population-otherText"],
+    monitoringMethods: entity?.accessMeasure_monitoringMethods?.map(
+      (method: AnyObject) =>
+        method.value === "Other, specify"
+          ? entity?.["accessMeasure_monitoringMethods-otherText"]
+          : method.value
+    ),
+    methodFrequency:
+      entity?.accessMeasure_oversightMethodFrequency?.[0].value !==
+      "Other, specify"
+        ? entity?.accessMeasure_oversightMethodFrequency?.[0].value
+        : entity?.["accessMeasure_oversightMethodFrequency-otherText"],
+  });
```

Comment:
This is on me for not cleaning this up, but is this ok? Should we abstract this out?